src/Microsoft.Health.ControlPlane.Core/Features/Exceptions/RoleNotFoundException.cs

```diff
@@ -0,0 +1,24 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using EnsureThat;
+
+namespace Microsoft.Health.ControlPlane.Core.Features.Exceptions
+{
+    public class RoleNotFoundException : ControlPlaneException
+    {
+        public RoleNotFoundException(string name)
+                : base(ValidateAndFormatMessage(name))
+        {
+        }
+
+        private static string ValidateAndFormatMessage(string name)
```

Comment:
Use the method from the base `ControlPlaneException` instead of recreating this static method. See `dentityProviderNotFoundException.cs` as an example.